modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/YauaaEnrichment.scala

```diff
@@ -63,14 +84,19 @@ object YauaaEnrichment extends ParseableEnrichment {
       case _ if s.length == 1 => s.toLowerCase
       case _ => s.charAt(0).toLower + s.substring(1)
     }
+
+  private def checkModel(model: Int): Either[String, Unit] =
+    if (SupportedContextModels.contains(model)) ().asRight
+    else s"""YAUAA enrichment context with model $model is not supported. Choose from: ${SupportedContextModels.mkString(", ")}""".asLeft
 }
 
 /**
  * Class for YAUAA enrichment, which tries to parse and analyze the user agent string
  * and extract as many relevant attributes as possible, like for example the device class.
+ * @param model version of yauaa_context produced, `1` and `2` are supported currently
  * @param cacheSize Amount of user agents already parsed that stay in cache for faster parsing.
  */
-final case class YauaaEnrichment(cacheSize: Option[Int]) extends Enrichment {
+final case class YauaaEnrichment(model: Int, cacheSize: Option[Int]) extends Enrichment {
```

Comment:
```suggestion
final case class YauaaEnrichment(cacheSize: Option[Int], model: Int = YauaaEnrichment.DefaultContextModel) extends Enrichment {
```

Reply:
I prefer to avoid defalut parameters as much as possible. In libs they're 100% a binary-compatibility breakage in apps often cause problems with type class derivation.

Reply:
I see, good to know!

Reply:
On the other hand FYI, in this case we'd already broke the binary compatibility by adding a parameter to case class: https://github.com/jatcwang/binary-compatibility-guide#avoid-using-case-classes

Reply:
...although case class parameter is also source incompatibility, so at least its very explicit.

Reply:
Thanks for the link, very interesting!